### PR TITLE
Update of the orogen-plugin documentation regarding API change

### PR DESCRIPTION
**Pull request by goldhoorn**
This fullfills
https://github.com/orocos-toolchain/orogen/issues/61

### CODE REVIEW

**Comment by doudou** [comment]
Use "Closes orocos-toolchain/orogen#61" to automatically close an issue when the PR is merged. I don't know if it can do it cross-repo though.

File: src/documentation/orogen/plugins.page [hunk 1]
```diff
@@ -175,3 +175,27 @@ and
...
+Adding user-defined classes to generation
+-----------------------------------------
+
+Plugins that want to register additional Classes during compilation
+could register a subfolders for the Cmake build system. These subfolders get 
+added to the compilation. The plugin developer has to make sure that the folder
+and a CMakeLists.txt within it has been generated during one of the above listed hooks.
+If this classes should be installed or linked against other parts, regular CMake commands must
+be used. The subfolders names must be export as a enumerable of string by the 
+```auto_gen_source_directory``` method of the plugin.
```
Comment (doudou) [inline src/documentation/orogen/plugins.page]: Isn't that "each_auto_gen_source_directory" ?

Comment (doudou) [inline src/documentation/orogen/plugins.page]: Also, you should probably add a small code example. The text will be hard to follow.

**Comment by goldhoorn** [comment]
Ok let's try this

File: src/documentation/orogen/plugins.page [hunk 2]
```diff
@@ -175,3 +175,41 @@ and
...
+could register a subfolders for the Cmake build system. These subfolders get 
+added to the compilation unit. The plugin developer has to make sure that the folder
+and a CMakeLists.txt within it has been generated during one of the above listed hooks.
+If this classes should be installed or linked against other parts, regular CMake commands must
+be used. The subfolders names must be export as a enumerable of string by the 
+```each_auto_gen_source_directory``` method of the plugin, example:
+
+
+~~~ ruby
+def each_auto_gen_source_directory(&block)
```
Comment (doudou) [inline src/documentation/orogen/plugins.page]: A much shorter version would be

~~~
def each_auto_gen_source_directory(&block)
    ["src1", "src2"].each(&block)
end
~~~

NOTE: don't add a &block parameter unless you need it, it's quite costly.

**Comment by goldhoorn** [comment]
1. Hmh, i thought about doing the short version but i assumed my one would be clearer (but even longer).
2. Your's is missing the enum_for stuff or is this not needed?
3. Why you mention the &block, the block is required here or miss i something?

I would prefer the longer version i have is your not strong against it...

**Comment by doudou** [comment]
Mine is more "rubyish" which means that non-ruby people don't completely understand what is going on. Basically, it's the way you can foward a each call "properly". Now, while I do agree that yours is easier to understand, it's also more cluttered. At the end, it boils down to whether the reader wants to learn about the language.

If he does, then my version is nice because it teaches him something

If he does not, then my version is also nice because it's very clear what he has to change :P

**Comment by goldhoorn** [comment]
whatever i don't care *updated*
